pulp_ostree/tests/functional/api/test_import.py

```diff
@@ -213,90 +175,102 @@ def single_ref_import(self, number_of_commits):
+            commit = ostree_content_commits_api_client.list(checksum=commits_to_check[i]).results[0]
+            assert commit.parent_commit is not None, commit
+            parent_commit = ostree_content_commits_api_client.read(commit.parent_commit)
+            assert parent_commit.checksum == commits_to_check[i - 1]
 
         # 10. publish the parsed commits
-        distribution_data = OstreeOstreeDistribution(**gen_distribution(repository=repo.pulp_href))
-        response = self.distributions_api.create(distribution_data)
-        distribution = monitor_task(response.task).created_resources[0]
-
-        ostree_repo_path = self.distributions_api.read(distribution).base_url
+        distribution = ostree_distribution_factory(repository=repo.pulp_href)
+        ostree_repo_path = ostree_distributions_api_client.read(distribution.pulp_href).base_url
 
         # 11. initialize a local OSTree repository and pull the content from Pulp
-        remote_name = init_local_repo_with_remote(self.repo_name2, ostree_repo_path)
-        self.addCleanup(shutil.rmtree, Path(self.repo_name2))
-        validate_repo_integrity(self.repo_name2, f"{remote_name}:foo", set(commits_to_check))
+        remote_name = init_local_repo_with_remote(repo_name2, ostree_repo_path)
+        validate_repo_integrity(repo_name2, f"{remote_name}:foo", set(commits_to_check))
 
         # 12. check if static deltas are being published
         for delta_path in deltas_paths:
-            response = self.client.using_handler(api.echo_handler).get(
-                urljoin(ostree_repo_path, delta_path)
-            )
-            self.assertEqual(response.status_code, 200)
-
-        monitor_task(self.distributions_api.delete(distribution).task)
-
-    def test_version_removal(self):
-        """Test the repository version removal functionality by removing two adjacent versions."""
-        self.single_ref_import(1)
-
-        repo_href = self.repositories_api.list().to_dict()["results"][0]["pulp_href"]
-        repo_version1_href = self.versions_api.read(f"{repo_href}versions/1/").pulp_href
-        repo_version2_href = self.versions_api.read(f"{repo_href}versions/2/").pulp_href
-
-        response = self.versions_api.delete(repo_version1_href)
-        monitor_task(response.task)
-        with self.assertRaises(requests.HTTPError) as exc:
-            self.client.get(repo_version1_href)
-        self.assertEqual(exc.exception.response.status_code, 404, repo_version1_href)
-
-        response = self.versions_api.delete(repo_version2_href)
-        monitor_task(response.task)
-        with self.assertRaises(requests.HTTPError) as exc:
-            self.client.get(repo_version2_href)
-        self.assertEqual(exc.exception.response.status_code, 404, repo_version2_href)
+            response = http_get(urljoin(ostree_repo_path, delta_path))
+            assert response
+
+        monitor_task(ostree_distributions_api_client.delete(distribution.pulp_href).task)
```

Comment:
This was added in the old test case to obliterate CI errors. I think we may want to return a distribution and repository from this function to make it possible to run these tests in parallel. The `test_version_removal` test should handle the removal of the distribution on its own. It was the only place where we wanted to get rid of distribution to be able to delete a repository version.